src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -973,6 +974,41 @@ using namespace ngraph::runtime;
     }
 }
 
+void runtime::cpu::CPU_ExternalFunction::propagate_in_place_input(
+    ngraph::descriptor::Output* output, std::string input_name)
+{
+    auto it = output;
+    auto propagate_further = false;
+    do
+    {
+        propagate_further = false;
+        for (auto input : it->get_inputs())
+        {
+            auto c_op = std::dynamic_pointer_cast<ngraph::op::Op>(input->get_node());
+            if (!c_op || c_op->is_output())
+            {
+                break;
+            }
+
+            if (auto op_annotations = c_op->get_op_annotations())
+            {
+                for (auto oi_pair : op_annotations->get_in_place_oi_pairs())
+                {
+                    if (oi_pair.input == input->get_index() && !oi_pair.destructive)
+                    {
+                        size_t output_index = oi_pair.output;
+                        auto& output_tensor = c_op->get_outputs().at(output_index).get_tensor();
+
+                        m_variable_name_map[output_tensor.get_name()] = input_name;
+                        it = &c_op->get_outputs().at(output_index);
+                        propagate_further = true;
```

Comment:
if we have more than one non-destructive users we will only propagate parameter's storage through one of them? I think we might want consider using a work stack here.

Reply:
good point. will address